### DISCUSSION TITLE
How do I tween a property of a shader

### DISCUSSION THREAD

**Discussion by THEkatinamicrowave** [CodenameCrew/CodenameEngine#437] [Q&A] [answered]
tried to shade
![image](https://github.com/user-attachments/assets/25912019-7719-463a-baf1-20510f1ab749)
I know this is for sure the wrong format but which one is
![image](https://github.com/user-attachments/assets/052b12f9-a41c-4cae-b310-8ea601b0159d)
for reference, here's the shader

```glsl
#pragma header

uniform vec3 targetColor;
uniform float intensity;

void main()
{
    vec4 originalColor = texture2D(bitmap, openfl_TextureCoordv);

    vec3 blendedColor = mix(originalColor.rgb, targetColor.rgb, originalColor.a * intensity);
    vec4 newColor = vec4(blendedColor, originalColor.a);

    gl_FragColor = newColor;
}

```

**Comment by NexIsDumb** [comment, marked as answer]
for the fields of a shader you gotta place them as strings, for ex:

```haxe
FlxTween.tween(bgShader, {"distortion": -10.0}, 30, {ease: FlxEase.circOut});

// orrr

FlxTween.tween(bgShader.data.distortion.value, {"[0]": -10.0}, 30, {ease: FlxEase.circOut});
```

**Reply by NeeEoo** [reply]
Fyi you need latest action build

also im noticing you are using a compiled build, you dont need to compile it, we have action builds that has the stuffs

**Reply by NexIsDumb** [reply]
> Fyi you need latest action build

ye that too

**Reply by THEkatinamicrowave** [reply]
yeah I installed that stuff immediately

**Reply by THEkatinamicrowave** [reply]
It's still erroring and crashing, except now it's just saying Object doesn't have property "[0]"
![image](https://github.com/user-attachments/assets/52569727-2e84-4f2e-a529-176f9d93fbd9)

**Reply by NeeEoo** [reply]
Run update.bat

**Reply by THEkatinamicrowave** [reply]
what does update.bat do? will any of my previous changes to the engine (I'm making a hardcoded mod) be undone?

**Reply by NeeEoo** [reply]
No also you dont need to hardcode in this engine, its made for softcoding. but it will only update the libraries, if you wanna be safe, make a copy of the .haxelib folder

**Reply by THEkatinamicrowave** [reply]
epic thanks
also ik I just kinda feel more comfortable hardcoding (it's weird because I started modding with psych engine)
if I make another mod I'll try softcoding out
